components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -1277,6 +1275,60 @@ public void revokeToken(String tokenId, String userId) throws IdentityOAuth2Exce
         return accessTokens;
     }
 
+    public Set<AccessTokenDO> getActiveDetailedTokensForConsumerKey(String consumerKey) throws IdentityOAuth2Exception {
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        Set<AccessTokenDO> activeDetailedTokens = new HashSet<>();
+        Map<String, AccessTokenDO> tokenMap = new HashMap<>();
+        try {
+            String sqlQuery = SQLQueries.GET_ACTIVE_DETAILS_FOR_CONSUMER_KEY;
+            ps = connection.prepareStatement(sqlQuery);
+            ps.setString(1, consumerKey);
+            ps.setString(2, OAuthConstants.TokenStates.TOKEN_STATE_ACTIVE);
+            rs = ps.executeQuery();
+            while (rs.next()) {
+                String token = rs.getString(2);
+                if (tokenMap.containsKey(token)) {
+                    AccessTokenDO tokenObj = tokenMap.get(token);
+                    String[] previousScope = tokenObj.getScope();
+                    String[] newSope = new String[tokenObj.getScope().length + 1];
+                    for (int size = 0; size < previousScope.length; size++) {
+                        newSope[size] = previousScope[size];
```

Comment:
Can use Array.copy

Reply:
Fixed with the commit c0b6f6f3b9cffe0410b3f26959d2a04be4c5131e